### PR TITLE
Github Actions (Integration test, unit test, PHP MD, Static code analysis)

### PR DESCRIPTION
**Pull request by Skullsneeze**
@ahmardi could you look into the failing unit tests? We might be able to skip unit testing on the order updates since they are covered by the integration tests I believe

### CODE REVIEW

File: Model/Checkout/PayOrder.php [hunk 1]
```diff
@@ -152,7 +153,7 @@ public function execute(int $orderId, string $amwalOrderId) : bool
             ->setLastOrderStatus($order->getStatus());
 
 
-        if ($amwalOrderStatus == 'success') {
+        if ($amwalOrderStatus == 'success' || $this->config->isIntegrationTestRun()) {
```
Comment (sameh-amwal) [inline Model/Checkout/PayOrder.php:156]: this seems a bit hacky. Have we considered mocking instead?

File: Test/Integration/IntegrationTestBase.php [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        'payment/amwal_payments/merchant_id' => 'sandbox-amwal-e09ee380-d8c7-4710-a6ab-c9b39c7ffd47',
+        'payment/amwal_payments/integration_test_run' => 1
+    ];
+
+    protected const MOCK_PRODUCT_SKU = 'amwal_simple';
+
+    protected const MOCK_REF_ID_DATA = [
+        RefIdDataInterface::IDENTIFIER => '100',
+        RefIdDataInterface::CUSTOMER_ID => '0',
+        RefIdDataInterface::TIMESTAMP => '1707916143'
```
Comment (sameh-amwal) [inline Test/Integration/IntegrationTestBase.php:42]: if we can do one recording of the api request with one full successful transaction we can use the recorded payload for replaying the test, right?

**Review by sameh-amwal — COMMENTED** [review]
This looks really great. Just had few comments and questions about using a replayed successful transaction.

File: Model/Checkout/PayOrder.php (hunk 1 above)
Comment (Skullsneeze) [reply Model/Checkout/PayOrder.php:156]: Yeah that might be something to look into indeed

File: Test/Integration/IntegrationTestBase.php (hunk 2 above)
Comment (Skullsneeze) [reply Test/Integration/IntegrationTestBase.php:42]: This is whats I tried initially, but the problem is that the reference ID is encrypted using the encryption key which is randomly generated with each test run, so the ref id would not match up and causes issues

Comment (sameh-amwal) [reply Test/Integration/IntegrationTestBase.php:42]: we can make it predictable by running using a preset integration set key each time and then using exactly the same key in our integration set run. the key for magento is stored unded `app/etc/env.php` like here
```
<?php
return [
    'backend' => [
        'frontName' => 'admin_1234'
    ],
    'remote_storage' => [
        'driver' => 'file'
    ],
    'queue' => [
        'consumers_wait_for_messages' => 1
    ],
    'crypt' => [
        'key' => 'ENCRYPTION_KEY'
    ],
```

**Review by sameh-amwal — APPROVED** [review]

**Pull request merged**